tests/foreman/longrun/test_inc_updates.py

```diff
@@ -212,3 +212,83 @@ def test_positive_noapply_api(
     )
+
+
+@pytest.mark.tier3
+def test_positive_incremental_update_time(module_target_sat, module_sca_manifest_org):
+    """Incremental update should not take a long time.
+
+    :id: a9cdcc58-2d10-42cf-8e24-f7bec3b79d6b
+
+    :steps:
+        1. Setup and sync rhel8 baseOS, rhst8, and rhsc8 repositories.
+        2. Create content view and add repos.
+        3. Publish a content view version with all content.
+        4. Using hammer, perform incremental update on that new version.
+        5. Publish the incremental version.
+
+    :expectedresults: Incremental update takes a short amount of time
+
+    :BZ: 2117760, 1829266
+
+    """
+    # rhel8 base os
+    rhel8_repo_id = module_target_sat.api_factory.enable_rhrepo_and_fetchid(
+        basearch=DEFAULT_ARCHITECTURE,
+        org_id=module_sca_manifest_org.id,
+        product=PRDS['rhel8'],
+        repo=REPOS['rhel8_bos']['name'],
+        reposet=REPOSET['rhel8_bos'],
+        releasever=REPOS['rhel8_bos']['releasever'],
+    )
+    rhel8_repo = module_target_sat.api.Repository(id=rhel8_repo_id).read()
+    assert rhel8_repo.sync(timeout=3000)['result'] == 'success'
+    # rh sat tools
+    rhst8_repo_id = module_target_sat.api_factory.enable_rhrepo_and_fetchid(
+        basearch=DEFAULT_ARCHITECTURE,
+        org_id=module_sca_manifest_org.id,
+        product=PRDS['rhel8'],
+        repo=REPOS['rhst8']['name'],
+        reposet=REPOSET['rhst8'],
+        releasever=None,
+    )
+    rhst8_repo = module_target_sat.api.Repository(id=rhst8_repo_id).read()
+    assert rhst8_repo.sync()['result'] == 'success'
+    # rh sat client
+    rhsc8_repo_id = module_target_sat.api_factory.enable_rhrepo_and_fetchid(
+        basearch=DEFAULT_ARCHITECTURE,
+        org_id=module_sca_manifest_org.id,
+        product=PRDS['rhel8'],
+        repo=REPOS['rhsclient8']['name'],
+        reposet=REPOSET['rhsclient8'],
+        releasever=None,
+    )
+    rhsc8_repo = module_target_sat.api.Repository(id=rhsc8_repo_id).read()
+    assert rhsc8_repo.sync()['result'] == 'success'
+    # create content view, add all repos
+    cv = module_target_sat.cli_factory.make_content_view(
+        {'organization-id': module_sca_manifest_org.id}
+    )
+    rhel_repos = [rhst8_repo_id, rhsc8_repo_id, rhel8_repo_id]
+    for repo_id in rhel_repos:
+        module_target_sat.cli.ContentView.add_repository(
+            {
+                'id': cv['id'],
+                'organization-id': module_sca_manifest_org.id,
+                'repository-id': repo_id,
+            }
+        )
+    # publish and fetch new CVV
+    module_target_sat.cli.ContentView.publish({'id': cv['id']})
+    content_view = module_target_sat.cli.ContentView.info({'id': cv['id']})
+    cvv = content_view['versions'][0]
+
+    # update incremental version of the CVV via hammer, publish it
+    timeout = 120  # seconds
+    start_time = datetime.utcnow() - timedelta(seconds=1)
```

Comment:
Btw, what is the current inc-update + publish time? Does the 120s have a safe margin?
Also wanted to ask about the `timedelta(seconds=1)` purpose. Some rounding?

Reply:
Locally I get about 20-30 seconds to update/create incremental version, and maybe 10-20 seconds to publish.
I will add some logging output of the time, and per @Griffin-Sullivan suggestion, think we should track the duration of each, incremental update & publish, individually.

As for `timedelta` rounding, I find this works best for asserting some satellite task time started on or after.
It's not looking for any tasks yet so wasn't quite needed, but the next push will find and `wait_for_tasks`.